src/sentry/static/sentry/app/components/events/eventDataSection.tsx

```diff
@@ -3,7 +3,26 @@ import React from 'react';
 import {t} from 'app/locale';
 import GuideAnchor from 'app/components/assistant/guideAnchor';
 
-class EventDataSection extends React.Component {
+const defaultProps = {
+  wrapTitle: true,
+  raw: false,
+  hideGuide: false,
+  toggleRaw: (_enable: boolean) => void 0,
+};
+
+type DefaultProps = Readonly<typeof defaultProps>;
+
+type Props = {
+  className?: string;
+  title: React.ReactText;
+  type: string;
+  wrapTitle: boolean;
+  toggleRaw: Function;
+  raw: boolean;
+  hideGuide: boolean;
+} & Partial<DefaultProps>;
```

Comment:
@markstory What's the advantage of doing `& Partial<DefaultProps>` when there are already types for the props?

Reply:
If I wasn't a muppet it lets you not have to redefine the types for default props. I'll clean that up.

Reply:
https://vignette.wikia.nocookie.net/disney/images/2/26/Walter.png/revision/latest?cb=20111013192451